File: Codewars/Typescript/7_kyu/Shortest Word.ts
// Description:
// Simple, given a string of words, return the length of the shortest word(s).

// String will never be empty and you do not need to account for different data types.

// My Solution:
export function findShort(s: string): number {
  let res: number = Number.POSITIVE_INFINITY;
  for (const c of s.split(" ")) {
    res = Math.min(c.length, res);
  }
  return res;
}

File: Codewars/Go/7_kyu/Shortest Word.go
// Description:
// Simple, given a string of words, return the length of the shortest word(s).

// String will never be empty and you do not need to account for different data types.

// My Solution:
package kata

import "strings"

func FindShort(s string) (res int) {
  words := strings.Split(s, " ")
  res = len(s)
  for _, word := range words {
    word_len := len(word)
    if word_len < res {
      res = word_len
    }
  }
  return res
}